### ISSUE TITLE
[Bug] Nondeterministic ModuleDict initialization order in MeanEmbeddingForecastLSTM due to unsorted set iteration

### ISSUE THREAD

**Issue by grey-nearing** [google-research/flood-forecasting#305]
## Summary

In `googlehydrology/modelzoo/mean_embedding_forecast_lstm.py` (lines 101–131), `hindcast_embeddings_fc` and `forecast_embeddings_fc` are constructed as `nn.ModuleDict` instances by comprehension over a Python `set`:

```python
self.hindcast_embeddings_fc = nn.ModuleDict(
    {
        name: self._create_fc(...)
        for name in set(
            self.config_data.hindcast_inputs_grouped.keys()
        ).difference(self.config_data.shared_groups)
    }
)
```

## Why This Breaks Reproducibility

Python randomizes string hash seeds across processes (`PYTHONHASHSEED`). Iterating over a `set` yields group names in an arbitrary, process-dependent order. Because `self._create_fc(...)` initializes layer weights by drawing sequentially from PyTorch's random number generator:
- In Process A, group `X` may be created first and consume random draws `1..N`, while group `Y` consumes `N+1..2N`.
- In Process B (with the **exact same `seed` in the YAML config**), group `Y` may be created first and consume draws `1..N`, while group `X` consumes `N+1..2N`.

Thus, two runs with the exact same configuration and random seed can initialize with different model weights across runs or machines.

A similar `set` iteration exists in `googlehydrology/datasetzoo/multimet.py` (`_get_products_and_bands_from_feature_strings`).

## Proposed Fix

1. Replace `set(...).difference(...)` with `sorted(set(...).difference(...))` (or preserve deterministic config key ordering via list comprehension `[k for k in ... if k not in ...]`) in `MeanEmbeddingForecastLSTM.__init__`.
2. Sort or preserve insertion order in `_get_products_and_bands_from_feature_strings`.
3. Add a unit test confirming that instantiating `MeanEmbeddingForecastLSTM` under different `PYTHONHASHSEED` values with the same `cfg.seed` produces bit-identical initial weights and `state_dict` key ordering.
